radix_sort/Cpp/radix_sort.cpp

```diff
@@ -0,0 +1,96 @@
+#include <iostream>
+#include <algorithm>
+#include <cmath>
+
+/*
+Utility function for Radix sort to sort given array for dth digit.
+This is essentially modified version of counting sort.
+
+Parameters:
+int a[] - Integer array a to be sorted.
+int n	- size of array a.
+int d	- Digit by which to sort the array with (0 based indexing).
+*/
+
+void radix_digit_util(int a[], int n, int d) {
+	if(a == NULL) return;
+
+	int exp = std::pow(10, d);
+
+	int *cnt = new int[n]; //array to store count of each occurences
+	int *out = new int[n]; //Out array to store sorted values
+
+	// Intitialise the count array to 0
+	for (int i = 0; i < n; ++i)
+		cnt[i] = 0;
+
+	// Count each occurences digit wise
+	for (int i = 0; i < n; ++i)
+		cnt[(a[i]/exp)%10]++;
+
+	// Cumulative count array
+	for (int i = 1; i < n; ++i)
+		cnt[i]+=cnt[i-1];
+
+	// Sort
+	for (int i = n - 1; i >= 0; i--) {
+	// Note - starting from n-1 to maintain "stable sort" property
+		out[cnt[(a[i]/exp)%10] - 1] = a[i];
+		cnt[(a[i]/exp)%10]--;
+	}
+
+	// Copy results back to orignal array
+	for (int i = 0; i < n; i++)
+		a[i] = out[i];
+}
+
+
+/*
+Radix sort implementation for sorting given array assuming the base 10.
+Note - // Works on Non-negative integer array only.
+
+Parameters:
+int a[] - Non-negative Integer array a to be sorted.
+int n	- size of array a.
+*/
+void radix_sort(int a[], int n) {
+	if(a == NULL) return;
+
+	// Get the iterator to max element for calculating maximum number of digits.
+	int * max_a_itr = std::max_element(a, a+n);
+	int max_a = *max_a_itr;
+
+	// Get maximum number of digits of all elements in array a
+	int max_d = 0;
+	while(max_a) {
+		max_d++;
+		max_a = max_a/10;
+	}
+
+	// Sort the array digit wise starting from lowest significant digit to highest.
+	for (int d = 0; d < max_d; d++)
+		radix_digit_util(a, n, d);
+}
+
+
+int main() {
+	std::ios_base::sync_with_stdio(false); //For faster io
+	int a[] = {1, 2, 9, 293, 41, 15, 52, 3, 121, 7, 1223, 3449, 15, 1 };
+	int n = sizeof(a)/sizeof(a[0]);
+
+	std::cout << "Input Array:\n";
+	for (int i = 0; i < n; i++) {
+		std::cout << a[i] << " ";
+	}
+	std::cout << "\n";
```

Comment:
use `using namespace std`

Reply:
@sangamcse I deliberately avoided that because it's considered not a good practice most of the times. See here - https://www.reddit.com/r/cpp_questions/comments/80ndnq/why_is_using_namespace_std_a_bad_practice/

I can still do it if you want. Please let me know. Thanks.

Reply:
Sorry to bug you on Saturday. But any update on this @sangamcse ?

Reply:
Sorry but this repo need https://github.com/NITSkmOS/Algorithms/pull/298#discussion_r223119805. In case it is required to change, we will create one issue for all cpp file and do it. But right now, we required what is mentioned in https://github.com/NITSkmOS/Algorithms/pull/298#discussion_r223119805